src/shared/fade-wrapper/fade-wrapper.tsx

```diff
@@ -0,0 +1,192 @@
+import { ChevronLeftIcon } from "@lifesg/react-icons/chevron-left";
+import { ChevronRightIcon } from "@lifesg/react-icons/chevron-right";
+import throttle from "lodash/throttle";
+import React, {
+    forwardRef,
+    useEffect,
+    useImperativeHandle,
+    useRef,
+    useState,
+} from "react";
+import { useResizeDetector } from "react-resize-detector";
+import {
+    Content,
+    Fade,
+    FadeIndicatorButton,
+    Wrapper,
+} from "./fade-wrapper.style";
+import { FadeColorSet, FadeWrapperProps, FadeWrapperRef } from "./types";
+
+const Component = (
+    {
+        children,
+        fadeColor,
+        fadePosition = "both",
+        showIndicator = false,
+        onResize,
+        ...otherProps
+    }: FadeWrapperProps,
+    ref: React.Ref<FadeWrapperRef>
+) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const [showFadeLeft, setShowFadeLeft] = useState<boolean>(
+        fadePosition === "left" || fadePosition === "both"
+    );
+    const [showFadeRight, setShowFadeRight] = useState<boolean>(
+        fadePosition === "right" || fadePosition === "both"
+    );
+
+    const wrapperRef = useRef<HTMLDivElement>(null);
+    const contentRef = useRef<HTMLDivElement>(null);
+
+    // To scroll left when wrapper resizes
+    useResizeDetector({
+        onResize: handleResize,
+        targetRef: wrapperRef,
+        refreshMode: "debounce",
+        refreshRate: 50,
+    });
+
+    useImperativeHandle(ref, () => {
+        return {
+            resize() {
+                handleResize();
+            },
+        };
+    });
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        const content = contentRef.current;
+
+        handleScroll();
+
+        window.addEventListener("resize", handleScroll);
+        if (content) {
+            content.addEventListener("scroll", throttle(handleScroll, 50));
```

Comment:
`throttle(handleScroll, 50)` returns a function, it should be declared as a const so that function reference is preserved for removeEventListener